lua/weapons/weapon_spraypaint/cl_init.lua

```diff
@@ -1,66 +1,576 @@
+    if IsValid(ply) then
+        local modelStr = ply:GetModel():sub(1, 17)
+        local isPony = modelStr == "models/ppm/player" or modelStr == "models/mlp/player" or modelStr == "models/cppm/playe"
+        local bn = isPony and "LrigScull" or "ValveBiped.Bip01_R_Hand"
         local bon = ply:LookupBone(bn) or 0
-        local opos = self:GetPos()
-        local oang = self:GetAngles()
         local bp, ba = ply:GetBonePosition(bon)
 
-        if (bp) then
+        if bp then
             opos = bp
         end
 
-        if (ba) then
+        if ba then
             oang = ba
         end
 
-        if ply:IsPony() then
-            --oang:RotateAroundAxis(oang:Forward(),90)
-            oang:RotateAroundAxis(oang:Up(), 90)
-            opos = opos + (oang:Up() * 0) + (oang:Right() * 8) + (oang:Forward() * 3)
+        if isPony then
+            opos = opos + (oang:Forward() * 6.4) + (oang:Right() * -1.8) + (oang:Up() * 0)
+            oang:RotateAroundAxis(oang:Right(), 80)
+            oang:RotateAroundAxis(oang:Forward(), 12)
+            oang:RotateAroundAxis(oang:Up(), 20)
         else
-            oang:RotateAroundAxis(oang:Up(), -90)
-            oang:RotateAroundAxis(oang:Forward(), 180)
-            opos = opos + (oang:Up() * -1) + (oang:Right() * 3) + (oang:Forward() * 2)
+            if (bn ~= 0) then
+                oang:RotateAroundAxis(oang:Forward(), 90)
+                oang:RotateAroundAxis(oang:Right(), 90)
+                oang:RotateAroundAxis(oang:Forward(), 90)
+                opos = opos + oang:Right() * 3.5
+                opos = opos + oang:Forward() * 2
+                opos = opos + oang:Up() * 0
+                oang:RotateAroundAxis(oang:Up(), -90)
+            end
+        end
+
+        if (isPony) then
+            if (horn) then
+                opos = opos - oang:Up() * 5
+            else
+                opos = opos + oang:Up() * 3
+            end
         end
 
         self:SetupBones()
-        local mrt = self:GetBoneMatrix(0)
+        matrix = self:GetBoneMatrix(0)
 
-        if (mrt) then
-            mrt:SetTranslation(opos)
-            mrt:SetAngles(oang)
-            mrt:SetScale(Vector(0.4, 0.4, 0.4))
-            self:SetBoneMatrix(0, mrt)
+        if matrix then
+            matrix:SetTranslation(opos)
+            matrix:SetAngles(oang)
+
+            if (query ~= true) then
+                self:SetBoneMatrix(0, matrix)
+            end
         end
     end
 
-    self:DrawModel()
+    if (query ~= true) then
+        render.MaterialOverride()
+        draw.NoTexture()
+        local cl = self:GetCustomColor()
+        render.SetColorModulation(cl.x, cl.y, cl.z)
+        render.SetBlend(1)
+        self:SetBodygroup(1, (self.CapOn and 0) or 1)
+        self:DrawModel()
+
+        for i = 0, 128 do
+            --print(i,self:GetBoneName(i))
+        end
+
+        render.SetColorModulation(1, 1, 1)
+    end
+
+    if (query == true) then return matrix, opos, oang end
```

Comment:
I think you need to make a GetWorldModelPosition function or something instead of this hack